Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3574,19 +3574,12 @@ namespace AssetProcessor
                         {
                             if (auto result = jobDependenciesDuplicateCheck.insert(jobDependency); !result.second)
                             {
-                                auto warningMessage = AZStd::string::format(
-                                    "Builder `%s` declared duplicate Job Dependencies for file `%s`.  Dependency: (`%s` `%s` `%s`).  Duplicates will be skipped.  "
-                                    "Please update the builder or content to remove the duplicates.",
-                                    builderInfo.m_name.c_str(),
-                                    actualRelativePath.toUtf8().constData(),
-                                    jobDependency.m_sourceFile.ToString().c_str(),
-                                    jobDependency.m_jobKey.c_str(),
-                                    jobDependency.m_platformIdentifier.c_str());
-
-                                AZ_Warning(AssetProcessor::DebugChannel, false, "%s", warningMessage.c_str());
-
-                                newJob.m_warnings.push_back(AZStd::move(warningMessage));
-
+                                // It is not an error or warning to supply the same job dependency
```

Comment:
I don't think there's actually currently a way to add "just messages' to the job to be emitted later during processjob, only warnings.  I'd have to add that functionality.  However, I don't think its necessary at all here, as per the comment.  IMO There is no need to issue any kind of log in a situation where its completely normal.

Reply:
To be more specific here, the ideal situation is that builders themselves are as simple as possible and have to do as little thinking, sorting, and filtering on a per-builder-basis as possible.  

This means that if AP can get a list of deps and have it be full of duplicates, its perfectly reasonable for AP to filter out the duplicates and continue, so that the builders themselves can be simpler and contain as little code as possible to deal with things like this (ie, the AP AssetBuilderSDK APIs should be as forgiving as possible).   

Which kinda implies that as long as the AP can make sense of the data coming from the builder, it should only be a message, error, or warning if something is *definitely* wrong.